util/constants.js

```diff
@@ -1,9 +1,10 @@
+import { ethers } from 'ethers'
+
 export const LvlV1Address = process.env.NEXT_PUBLIC_LVL_V1_CONTRACT_ADDRESS
 export const InfuraId = process.env.NEXT_PUBLIC_INFURA_ID
-export const NetworkId = process.env.NEXT_PUBLIC_NETWORK_ID
-
-const NETWORK_NAMES = {
-  '0x1': 'Mainnet',
-  '0x4': 'Rinkeby',
+export const Network = {
+  id: Number(process.env.NEXT_PUBLIC_NETWORK_ID),
+  hexId: `0x${Number(process.env.NEXT_PUBLIC_NETWORK_ID).toString(16)}`,
+  name: ethers.providers.getNetwork(Number(process.env.NEXT_PUBLIC_NETWORK_ID))
+    .name,
 }
```

Comment:
Trying to load as much as possible from 3rd party libraries so we don't have to update all these constants across projects frequently.